src/compiler/performance.ts

```diff
@@ -1,107 +1,110 @@
-/*@internal*/
 namespace ts {
+    declare const performance: { now?(): number } | undefined;
+    /** Gets a timestamp with (at least) ms resolution */
+    export const timestamp = typeof performance !== "undefined" && performance.now ? performance.now : Date.now ? Date.now : () => +(new Date());
+}
+
+/*@internal*/
+namespace ts.performance {
     /** Performance measurements for the compiler. */
-    export namespace performance {
-        declare const onProfilerEvent: { (markName: string): void; profiler: boolean; };
-        declare const performance: { now?(): number } | undefined;
-        let profilerEvent: (markName: string) => void;
-        let markInternal: () => number;
-        let counters: Map<number>;
-        let measures: Map<number>;
+    declare const onProfilerEvent: { (markName: string): void; profiler: boolean; };
+    let profilerEvent: (markName: string) => void;
+    let markInternal: () => number;

```

Comment:
It shouldn't. The whole reason we switched to this approach was to avoid constantly mutating `ts` with our timestamps so that it is less polymorphic so that calls to things like timestamp can be inlined.